src/antares/model/study.py

```diff
@@ -194,6 +212,11 @@ def service(self) -> BaseStudyService:
         return self._study_service
 
     def read_areas(self) -> list[Area]:
+        area_list = self._area_service.read_areas()
```

Comment:
You don't update the self._areas here, you need to do it. Also you do 2 calls to read which is time consuming
The code should look like this
```python
def read_areas(self) -> list[Area]:
    area_list = self._area_service.read_areas()
    self._areas = {area.id: area for area in area_list}
    return area_list
```

Also could you add a little doc inside this method saying it's used to syncronize the internal study object with the real object written inside a study